cmd/s2i/main.go

```diff
@@ -146,10 +146,6 @@ $ s2i build . centos/ruby-22-centos7 hello-world-app
 
 			glog.V(2).Infof("\n%s\n", describe.DescribeConfig(cfg))
 
-			if !docker.IsReachable(cfg) {
```

Comment:
@jasinner - thanks for the PR - minimally it is instructive since it lets us know what you did to work around the issue.

As to the precise fix, IMO, given the processing of those environment variables [here](https://github.com/openshift/source-to-image/blob/master/pkg/docker/util.go#L344-L356), I'm curious why the set up [here](https://github.com/openshift/source-to-image/blob/master/pkg/docker/docker.go#L287-L325) did not provide favorable results.

I think the s2i team should use https://github.com/openshift/source-to-image/issues/604 to investigate that and come up with a fix that allows `IsReachable` to succeed with TLS enabled.

@bparees thoughts?

Also, it is possible that the back off of k8s in favor of direct engine-api access could have some bearing here.